File: in_Java/0199 Binary Tree Right Side View.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        Queue<TreeNode> q = new LinkedList<>();
        if(root == null) return res;
        q.add(root);
        int len;
        TreeNode cur;
        while(q.size() != 0){
            len = q.size();
            while(len != 0){
                cur = q.poll();
                if(len == 1) res.add(cur.val);
                if(cur.left != null) q.add(cur.left);
                if(cur.right != null) q.add(cur.right);
                len--;
            }
        }
        return res;
    }
}

File: in_Python/0199 Binary Tree Right Side View.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def rightSideView(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        q = []
        res = []
        if root == None:
            return res
        q.append(root)
        while len(q) != 0:
            n = len(q)
            cur = []
            for i in range(n):
                h = q.pop(0)
                if h.left != None:
                    q.append(h.left)
                if h.right != None:
                    q.append(h.right)
                cur.append(h.val)
            res.append(cur[-1])
        return res
        